ROADMAP.md

```diff
@@ -0,0 +1,277 @@
+# rust-libp2p Roadmap
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [rust-libp2p Roadmap](#rust-libp2p-roadmap)
+    - [Testground abstraction](#testground-abstraction)
+    - [TLS support](#tls-support)
+    - [QUIC support](#quic-support)
+    - [WebRTC support (browser-to-server)](#webrtc-support-browser-to-server)
+    - [Release process](#release-process)
+    - [Refactor event handling in `Swarm`](#refactor-event-handling-in-swarm)
+    - [Cross Behaviour communication](#cross-behaviour-communication)
+    - [swarm/handler: Decouple ConnectionHandler from {In,Out}boundUpgrade #2863](#swarmhandler-decouple-connectionhandler-from-inoutboundupgrade-2863)
+    - [Generic connection management](#generic-connection-management)
+    - [Kademlia efficient querying](#kademlia-efficient-querying)
+    - [Kademlia client mode](#kademlia-client-mode)
+    - [Hole punching](#hole-punching)
+    - [Streaming response protocol aka. the ultimate escape hatch](#streaming-response-protocol-aka-the-ultimate-escape-hatch)
+    - [Improved WASM support](#improved-wasm-support)
+    - [Muxer handshake optimization](#muxer-handshake-optimization)
+    - [Bitswap implementation](#bitswap-implementation)
+    - [WebTransport](#webtransport)
+
+<!-- markdown-toc end -->
+
+## Testground abstraction
+
+Status: todo
+
```

Comment:
The roadmap items already express relative priority through the order in which the items are listed. What value would an additional _priority_ property per roadmap item bring? In other words, what questions would this property help answer?

> It is hard to discern which items are most critical vs which are nice to haves.

I consider all items on this roadmap important, i.e. worth doing eventually.

Reply:
> The roadmap items already express relative priority through the order in which the items are listed. What value would an additional priority property per roadmap item bring? In other words, what questions would this property help answer?

I see since that's the case then I agree, an additional priority property would not bring any additional value.

But it wasn't evident that it was listed in order of descending priority when I first reviewed it. The addition of `Target completion:` helps.
My suggestion is to add a blurb at the top of this doc and say as much, this will help new readers out.